### ISSUE TITLE
[$4000] Android - Chat - Empty chat (black screen) is displayed instead of messages

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17870]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Go https://staging.new.expensify.com/
, login, copy URL of any chat, save it in Email (or slack, or Telegram) for later to tap
2. Open Android NewDot app (could be logged in or logged out)
3. Tap deep link from step 1

## Expected Result:
Chat should be open with previous history.

## Actual Result:
Empty chat (like black screen) is displayed instead of messages

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.4.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/233865409-7349c550-cb89-44e5-81a9-eaf1bc2072ae.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018e845ce5e7422dc1</li>
        <li>Upwork Job ID: 1654063892620099584</li>
        <li>Last Price Increase: 2023-05-19</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @maddylewis (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by maddylewis** [comment]
oopsie! let me try repo'ing this one now!

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by dylanexpensify** [comment]
Woot! gotchu!

**Comment by dylanexpensify** [comment]
Hmmm can't repro. @kbecciv can you confirm if this is still occurring?

**Comment by dylanexpensify** [comment]
bump @kbecciv

**Comment by kbecciv** [comment]
Hello @dylanexpensify! Checking, will update you shortly.

**Comment by kbecciv** [comment]
@dylanexpensify Issue is reproductible with build 1.3.8.8

https://user-images.githubusercontent.com/93399543/235657139-41f01c89-0fe2-46f0-ba1a-19b620169d72.mp4

**Comment by dylanexpensify** [comment]
Thanks @kbecciv! Didn't get to this today due to focusing on ECUK and CPM tasks! Will review tomorrow!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~018e845ce5e7422dc1

**Comment by MelvinBot** [comment]
Current assignee @dylanexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @0xmiroslav (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @alex-mechler (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by therealsujitk** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The list of comments don't show at random times, after the skeleton animation it just shows a blank page.

**Note:** This issue doesn't happen only when you click on a URL to open a chat, it happens randomly when you go back and fourth chats. 


https://user-images.githubusercontent.com/60378235/236403601-61975a73-42a8-4580-aab4-aae69418eae2.mp4

### What is the root cause of that problem?

The cause of this problem is the `react-native-reanimated` package. To display the comments animation style has been added to animate the opacity from 0 to 1 in 100ms. This animation seems to fail at random times keeping the opacity set to 0.

https://github.com/Expensify/App/blob/3585329ec6fe8a8bed127fc44b06ff698380a440/src/pages/home/report/ReportActionsList.js#L79-L85

[A similar issue](https://github.com/software-mansion/react-native-reanimated/issues/3607) was also opened by someone else in the [react-native-reanimated](https://github.com/software-mansion/react-native-reanimated) repository.

### What changes do you think we should make in order to solve the problem?


The requirement here is a fairly simple fade animation, which is why I suggest using react natives built-in Animated API which works quite well and doesn't have this issue (I've tested).

### What alternative solutions did you explore? (Optional)

- The issue seems to go away when you increase the animation duration, however this solution is not foolproof.
- [Other react native animation libraries](https://blog.bitsrc.io/top-5-animation-libraries-in-react-native-d00ec8ddfc8d) can be used, however similar or worse issues might show up.

**Comment by dylanexpensify** [comment]
Didn't get to this today due to prioritizing Payroll, CPM, ECUK items! Will review Monday!

**Comment by dylanexpensify** [comment]
@0xmiroslav can we get reviews on proposals and an update today please?

**Comment by therealsujitk** [comment]
@0xmiroslav found it! https://github.com/Expensify/App/pull/16613. And it was previously animated the way I suggested in my proposal, that could be when this issue started occurring (March 28, few days over a month ago).

**Comment by therealsujitk** [comment]
@0xmiroslav no no, this refactoring also involved migrating the animation to use the `react-native-reanimated` package which caused this issue. Nothing to do with the refactor.

**Comment by therealsujitk** [comment]
@0xmiroslav I just checked by upgrading the package, recompiling the application, clearing metro cache, recompiling again, and the issue still exists in `react-native-reanimated@3.1.0`.

P.S. sorry for the delay, my laptop is really slow and takes a while to build.

**Comment by therealsujitk** [comment]
@0xmiroslav I mentioned a method to reproduce this in my proposal by going back-and-fourth chats. Also, I believe this issue might have something to do with the phone's performance, so if you have a slightly older phone or if you can temporarily reduce your emulator's RAM maybe that would help?

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by alex-mechler** [comment]
@0xmiroslav have you been able to get a reproduction with @therealsujitk's suggestions?

**Comment by therealsujitk** [comment]
I'm trying to figure out why the first issue is happening. It's not `react-reanimated` since we use a package called `react-content-loader` for this. I'm honestly not sure if this can be fixed, could be  related to  https://github.com/danilowoz/react-content-loader/issues/257.

I'm pretty sure this also has to do with the phone's performance again. The android application (mWeb Android too) has always seemed very laggy to me, every other app on my phone is quite smooth so I'm not sure why the Expensify app is so slow. It seems to work fine on iOS.

If remember correctly, a few years back I was reading a post by Discord of them saying they used react-native and electron but they didn't release their android app because it was not smooth enough (or something), I wonder how they improved it.

**Comment by alex-mechler** [comment]
@0xmiroslav we should fix both. If we have data we are currently loading, we should show the skeleton. If we have already loaded the data, we should be showing the chat. We should not be showing a blank page in either situation

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by dylanexpensify** [comment]
@0xmiroslav to confirm, we have no pending or accepted proposals, correct?

**Comment by dylanexpensify** [comment]
Ok, doubling to get some steam

**Comment by dylanexpensify** [comment]
Updated again to increase urgency

**Comment by therealsujitk** [comment]
@0xmiroslav I was wondering if maybe `InteractionManager.runAfterInteractions()` could help with this issue, I just saw this used somewhere else and when I googled what it was it seemed to fit the issue (I think), however I'm not sure if it'll actually work or if it can even be used here.

Here's the funny part, I can't reproduce this on my phone anymore :sweat_smile:, I'm not sure what happened to this project in the past week but suddenly the app works lightning fast, all animations and everything in 120Hz :exploding_head:.

**Comment by dylanexpensify** [comment]
@alex-mechler lmk what ETA we have for squashing this now that it's internal!

**Comment by alex-mechler** [comment]
No ETA at the moment, have some other internal / Manual Requests issues that are taking priority

**Comment by alex-mechler** [comment]
> Here's the funny part, I can't reproduce this on my phone anymore 😅, I'm not sure what happened to this project in the past week but suddenly the app works lightning fast, all animations and everything in 120Hz 🤯.

We did reanable Hardware Acceleration for android, I wonder if that is related

**Comment by therealsujitk** [comment]
@0xmiroslav from your video it looks like the skeleton is being loaded as well, but after a delay.

**Comment by alex-mechler** [comment]
No update here, been ooo / working on manual request polish

**Comment by alex-mechler** [comment]
Same as last time, no update, still working on manual request polish

**Comment by puneetlath** [comment]
That's a good idea @0xmiroslav. I'm going to do that.

**Comment by alex-mechler** [comment]
Great idea, thanks for asking Puneet!

**Comment by Piotrfj** [comment]
Hi I'm Piotr from Callstack - expert contributor group - will start investigation in this area

**Comment by Piotrfj** [comment]
Update: Im continuing to investigate the issue, nothing at this moment was found

**Comment by Piotrfj** [comment]
Update: The issue is hardly reproducible, after reproducing I got an error with reanimated. After searching for the error I have found that the issue is reported to the authors and its not yet fixed. Gonna try to bump up the version locally and see if i still can get the error.

**Comment by alex-mechler** [comment]
Any update here @Piotrfj?

**Comment by dylanexpensify** [comment]
not overdue, just commented!

**Comment by Piotrfj** [comment]
Update: Tested the solution mentioned by @WikusKriek, the bug did not appear anymore, cannot be 100% sure thought because the bug is really hard to reproduce.

**Comment by quinthar** [comment]
Just confirming that we are processing this as a regression right? We are going to do a root cause analysis to determine how specifically this broke, and make sure this doesn't happen again?

**Comment by dylanexpensify** [comment]
Agree! @0xmiroslav can you give us an update here please?

**Comment by Piotrfj** [comment]
@0xmiroslav As mentioned, after applying the solution of @WikusKriek I did not encounter the bug again, cannot be 100% sure because the bug is hardly reproducible. Should i write a proposal using this solution, or it is enough that some other fix also fixes this bug?

**Comment by hannojg** [comment]
Hey, Hanno here from Margelo, external contributor. As this was flagged as high priority issue we will look into this as well!

There [are two issues:](https://github.com/Expensify/App/issues/17870#issuecomment-1546993100)

> - shows empty chat forever after data fully loaded
> - shows empty chat briefly instead of skeleton before loading report data

And for "shows empty chat forever" we already [have a proposal](https://github.com/Expensify/App/issues/17870#issuecomment-1535870254):

> " [...] using react natives built-in Animated API"

(imo it would be interesting to find the root cause of the issue why it sometimes doesn't work using reanimated, however, given the urgency I'd say that changing it back to react-native's Animated API is fine at the minute)

So, we only need to look at "shows empty chat briefly instead of skeleton before loading report data", for which WikusKriek [left a hint here](https://github.com/Expensify/App/issues/17870#issuecomment-1579314835): 

> To solve the loading skeletop page flashing issue, we can change the shouldShow prop of the FullPageNotFoundView:https://github.com/Expensify/App/blob/dcb6c2ca9950602c48e869380db120d3c0b79099/src/pages/home/ReportScreen.js#L270
to this:
```jsx
shouldShow={(!this.props.report.reportID) || shouldHideReport}
```

I will look into fixing the second issue, which we can combine with the first solution.

**Comment by hannojg** [comment]
I just want to add another finding to this, I think when the app's JS thread is dead, it just not fast enough to keep up with rendering the skeleton 😅 :


https://github.com/Expensify/App/assets/16821682/03dec6c9-c7b0-49a1-8d7b-9cace26ee86f

**Comment by hannojg** [comment]
Also I don't think that in the case where it gets stuck we are showing the `FullPageNotFoundView` view, as this one would look like this, so I don't think its related @WikusKriek 

![Simulator Screen Shot - iPhone 14 - 2023-06-20 at 18 20 49](https://github.com/Expensify/App/assets/16821682/4187cda6-1711-4ec1-a65c-fbfe1fe79ab3)

**Comment by michaelhaxhiu** [comment]
Removing AlexM's assignment and assigning a new CME.

**Comment by PauloGasparSv** [comment]
Sorry for the delay but I still have to catch up with the discussion here!

Will pick this up and try to consolidate all the info we have so far on Monday

**Comment by quinthar** [comment]
I'm still hitting this all the time. Daily.  I'm on v1.3.31-2.  Can we add some logging or can I do anything to help whenever I hit this?

**Comment by quinthar** [comment]
Just reproduced it again.  If I tap around a bit, eventually it happens.  This time I tapped on a chat, it loaded fine, I went to the LHN, then tapped on that same chat, and now it's just a blank screen.   How can I help diagnose this?  Can someone check my logs?

**Comment by quinthar** [comment]
Specifically what happens is I:

1. Open the chat (a group DM to 2 others), it worked
2. Tap <
3. Tap on the chat
4. It shows the "skeleton" briefly
5. Then the skeleton disappears and the page is just blank -- shows the header and compose, but the content area is blank
6. Goto 2; repeat endlessly

**Comment by quinthar** [comment]
Actually it seems i can reproduce this very reliably with almost every chat.  It doesn't always break the 2nd time I open it, but most of the time.

**Comment by hannojg** [comment]
@quinthar Lets do two things:

## 1. Investigate if we can find the issue in the logs

I build a release version of the app that still has logging enabled. You'll need to install that version and then connect your mobile to your computer to get us the logs, as outlined here:

1. Download the app and install it on your phone [using this build](https://wetransfer.com/downloads/183defc1e623a2b5aab2ede534f5112020230626091235/09eb8276960e4dfffc36d5f0c8c4559920230626091251/8925fa)
2. Connect your mobile to your PC
3. Make sure you have the developer mode enabled on your phone and turned on USB debugging
4. Make sure you have the `adb` installed on your machine (I think a while ago we installed it on your machine)
5. Run `adb devices` and make sure you see your mobile
6. Run `adb logcat > android_logs.txt`
7. Open the app on your phone and cause the bug
8. Then finish the log capturing by pressing CTRL + C
9. Share the android_logs.txt file with us
10. **Pro tipp:** Run `adb bugreport` and share the zip file with us. If there is anything strange in the system internals going on we will find it there (I don't think that's the case but lets be sure)

## 2. Try what happens when we remove any reanimated usage in the animation for the skeleton and content

Earlier we were suspecting the reanimated code to be the problem for this, but none of us could reliably reproduce to verify.
I created a second build that has REA disabled in this part of the code.
Please run this app version and tell us whether the bug is still happening (note: the loading UI might be broken, but we just want to know if in the end the chats are loading or not)

https://we.tl/t-LrJ8cvcRed

Thank you for taking the time to get us the information we need 🙌

**Comment by dylanexpensify** [comment]
Didn't get to today, but will tackle tomorrow!

**Comment by muttmuure** [comment]
👋 I have found some logs from when David reported this: 

[[hmmm] [PusherConnectionManager] Channels Error 1006 ~~ error: '[type: 'PusherError' data: '[code: '1006' message: 'Server lost session']']' userAgent: 'okhttp/4.11.0'](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id:(%227da00c9fc8d3c398-SEA%22)+AND+timestamp:[2023-06-19T23:55:58.828Z+TO+2023-06-20T01:55:58.828Z]&index=logs_expensify-022887)


[[info] [Network] API request error: Request canceled ~~ message: 'Aborted' request: '[command: 'ReconnectApp' data: '[policyIDList: '[0: '01DC7E46150451C4' 1: '0D00F2B7A62842CD' 2: '137DA25D273F2423' 3: '1F92C722664305E0' 4: '2E7DFDED1F82B2F0' 5: '2FCADD02D1DF4D8F' 6: '3A89327178065942' 7: '47A7FA5FAFEDDD63' 8: '4E8D009A6CC35A3E' 9: '5FBBD03016000000' 10: '6853528C70F040C9' 11: '72BEC40917763029' 12: '73860F80028ECDF1' 13: '7A35FD885AE104E1' 14: '7DCB3E465D20C5CE' 15: '83C1DCD27E5C95DD' 16: 'A55A7962A1F001A4' 17: 'A5F944A401046336' 18: 'ABF8D1BA1FBC218A' 19: 'B99FD813110CBCD3' 20: 'BA1D968239FA073C' 21: 'BD42B7BBB0AFA612' 22: 'BF5C59CAEE213690' 23: 'CD67AE8AC59D8886' 24: 'F11FF01563221872']' appversion: '1.3.29-0' apiRequestType: 'write' pusherSocketID: '475332.2966989' shouldRetry: '1' canCancel: '1']' successData: '[0: '[onyxMethod: 'merge' key: 'isLoadingReportData' value: '']']' failureData: '[0: '[onyxMethod: 'merge' key: 'isLoadingReportData' value: '']']']' userAgent: 'okhttp/4.11.0'](https://www.expensify.com/_devportal/tools/logSearch/#query=request_id%3A(%227da014034e80f8d9-SEA%22)%20AND%20timestamp%3A%5B2023-06-20T00%3A01%3A01.405Z%20TO%202023-06-20T02%3A01%3A01.405Z%5D&index=_all)

**Comment by PauloGasparSv** [comment]
@dylanexpensify @quinthar bump on @hannojg's [logs request](https://github.com/Expensify/App/issues/17870#issuecomment-1607077541)? 

> 1. shows empty chat forever after data fully loaded
> 2. shows empty chat briefly instead of skeleton before loading report data

Hey @hannojg do you have any updates on item 2 as well?
And are you still unable to reproduce item 1 reliably?

**Comment by dylanexpensify** [comment]
@hannojg can we get an update please? Will defer to @quinthar on the logs request

**Comment by hannojg** [comment]
The download links for David expired, need to post new / permanent ones!

I haven't been able to reproduce that issue locally yet, so I don't have a proposal yet. I have this on my plate for this week, and will try to get to the root of it. With the new logs from @muttmuure I may have a few new places to look 😊

**Comment by hannojg** [comment]
We just made a bunch of proposals to make loading data from onyx more efficient + improve first rendering. My gut tells me that these proposals will likely help with this issue (its still a pity that we can't really reproduce the issue :( )

https://expensify.slack.com/archives/C01GTK53T8Q/p1688580518417739

Once I have issues up for the proposals I will list them here, so we can put this issue on hold for those other issues 😊

**Comment by dylanexpensify** [comment]
@hannojg any updates? 🙇‍♂️

**Comment by hannojg** [comment]
By tomorrow we should have the issue, will let you know then!

**Comment by hannojg** [comment]
@dylanexpensify Here you go:

- https://github.com/Expensify/App/issues/22644

**Comment by PauloGasparSv** [comment]
Putting this on HOLD for https://github.com/Expensify/App/issues/22644 as we mentioned we would do [here](https://github.com/Expensify/App/issues/17870#issuecomment-1623360308)!

**Comment by PauloGasparSv** [comment]
Hey, I'm going OOO for 2 weeks (starting on July 17th, back on July 31st) so please feel free to re-assign this to another internal engineer!

Not doing that now since this is on hold : )

**Comment by dylanexpensify** [comment]
on hold!

**Comment by dylanexpensify** [comment]
on hold

**Comment by hannojg** [comment]
We are just working on getting the PR into the app:

https://github.com/Expensify/App/pull/23749

**Comment by PauloGasparSv** [comment]
> As reported by David, who mainly had this issue, the aforementioned PR is fixing the issue 🥳

I believe we can close this now that https://github.com/Expensify/App/pull/23749 was deployed to prod then right?

**Comment by hannojg** [comment]
Yes, we are now waiting for this PR:

- https://github.com/Expensify/App/pull/24041

**Comment by dylanexpensify** [comment]
Nice one!

**Comment by dylanexpensify** [comment]
Still in hold? @hannojg

**Comment by dylanexpensify** [comment]
amazing!

**Comment by dylanexpensify** [comment]
reviewing today!

**Comment by dylanexpensify** [comment]
@Piotrfj when are we getting the PR raised?

**Comment by dylanexpensify** [comment]
Reviewing tomorrow! Got caught up with higher priority items

**Comment by dylanexpensify** [comment]
@0xmiroslav mind confirming @hannojg's request?

**Comment by dylanexpensify** [comment]
Bump @0xmiroslav

**Comment by dylanexpensify** [comment]
updating today!

**Comment by dylanexpensify** [comment]
@hannojg to review responses 🙇‍♂️

**Comment by dylanexpensify** [comment]
@0xmiroslav any chance of reproing? Adding new engineer

**Comment by hannojg** [comment]
I think it should be a new issue for "showing a blank before showing the skeleton loading" (which we are currently already working on by fixing the performance for opening a report).

And the issue that a chat never loads isn't reproducible anymore since the onyx changes we made, so I think this issue is done.

**Comment by dylanexpensify** [comment]
Reviewing today while in the air! Currently taxi'd, will get to once I can hop on wifi!

**Comment by dylanexpensify** [comment]
Perfect, sounds like we're good to close out? @hannojg @0xmiroslav

**Comment by dylanexpensify** [comment]
Bump @quinthar

**Comment by mallenexpensify** [comment]
I'm on the #expensify-open-source weekly update, assigned to @quinthar for 👀 to see if he's able to reproduce.

**Comment by luacmartins** [comment]
Just waiting on confirmation that this is still happening

**Comment by quinthar** [comment]
Oh hey I'm sorry, I didn't see these bumps as it wasn't assigned to me.  I haven't seen this situation in a long time, so I think we're good!

**Comment by luacmartins** [comment]
Awesome! We're good to close then. Thanks everyone!
